npe2/manifest/schema.py

```diff
@@ -147,12 +147,27 @@ def _check_engine_version(cls, values: dict) -> dict:
             )
         return values
 
+    @validator("display_name")
+    def validate_display_name(cls, v):
+
+        regex = r"^[^\W_][\w -~]{1,38}[^\W_]$"
+        if not bool(re.match(regex, v)):
+            raise ValueError(
+                f"{v} is not a valid display_name.  The display_name must "
+                "be 3-40 characters long, containing printable word characters, "
+                "and must not begin or end with an underscore, white space, or "
+                "non-word character."
+            )
+        return v
+
     @root_validator
     def _validate_root(cls, values: dict) -> dict:
         invalid_commands = []
+        invalid_python_names = []
```

Comment:
```suggestion
```

Reply:
i don't think we need this anymore. also suggested removing a related block below.